converse.js

```diff
@@ -634,11 +635,35 @@
             },this), 200));
         };
 
+        this.ping = function (jid, timeout, success, error){
+            if (typeof jid === 'undefined'     || jid == null)     { jid = this.bare_jid; }
+                if (typeof timeout === 'undefined' || timeout == null) { timeout = 45; }
+                if (typeof success === 'undefined' ) { success = null; }
+                if (typeof error === 'undefined' ) { error = null; }
+                if (this.connection) this.connection.ping.ping( jid, success, error, timeout );
+        };
```

Comment:
Normaly , the connection would change and onConnStatusChanged should already be called.
my opinion is there is no need to force error callback to do this checks.
There is also servers that don't manage ping and will send errors but it will keep the connection.